File: src/PizzaPlus/XmlReader.java
package PizzaPlus;
import javax.xml.parsers.SAXParser;
import javax.xml.parsers.SAXParserFactory;
import java.io.File;

public class XmlReader {
    private final String inputFileName;
    private XmlHandler toppingXmlHandler;
    private Toppings[] toppings;

    public XmlReader(String inputFileName) {
        this.inputFileName = inputFileName;

        try {
            // Declare a File object
            File inputFile = new File(inputFileName);

            // Create a SAX parser factory and parser
            SAXParserFactory saxParserFactory = SAXParserFactory.newInstance();
            SAXParser saxParser = saxParserFactory.newSAXParser();

            if (inputFileName.equals("toppings.xml")){
                toppingXmlHandler = new XmlHandler();
                saxParser.parse(inputFile, toppingXmlHandler);
            }

        } catch (Exception e) {

            System.out.println("Exception: " + e.getMessage());

        }
    }

    public Toppings[] getToppingsList(){
        // Get the ArrayList of Orders built by the handler and return an array
        Toppings[] list = new Toppings[toppingXmlHandler.getToppings().size()];
        list = toppingXmlHandler.getToppings().toArray(list);
        return list;
    }
}

File: src/PizzaPlus/XmlReader.kt
package PizzaPlus

import PizzaPlus.Customer.*
import java.io.File
import java.lang.Exception
import javax.xml.parsers.SAXParserFactory

class XmlReader(private val inputFileName: String) {
    private var toppingXmlHandler: XmlHandler? = null
    private var toppings: Array<Toppings>? = null


    //var itemsOrdered = ArrayList<Pizza?>()

    // Get the ArrayList of Orders built by the handler and return an array
//    val toppingsList: Array<Toppings>
////        get() {
////            // Get the ArrayList of Orders built by the handler and return an array
//////            var list = ArrayList<Toppings>(toppingXmlHandler?.getToppings().size())
//////            list = toppingXmlHandler?.getToppings().toArray(list)
//////            return list
////        }

    init {
        try {
            // Declare a File object
            val inputFile = File(inputFileName)

            // Create a SAX parser factory and parser
            val saxParserFactory = SAXParserFactory.newInstance()
            val saxParser = saxParserFactory.newSAXParser()
            if (inputFileName == "toppings.xml") {
                toppingXmlHandler = XmlHandler()
                saxParser.parse(inputFile, toppingXmlHandler)
            }
        } catch (e: Exception) {
            println("Exception: " + e.message)
        }
    }
}